pmac/src/double.rs

```diff
@@ -0,0 +1,140 @@
+use block_cipher_trait::generic_array::typenum::{U8, U16, U32};
+use super::Block;
+
+use core::mem;
+
+const C64: u64 = 0b1_1011;
+const C128: u64 = 0b1000_0111;
+const C256: u64 = 0b100_0010_0101;
+
+/// Some block operations over Galois Field.
+///
+/// This trait is implemented for 64, 128 and 256 bit block sizes.
+/// IMPORTANT: block must be aligned
+pub trait Doublable {
```

Comment:
Any thoughts on splitting this out into a separate crate? I need it for SIV too. It'd be nice to have a single implementation CMAC, PMAC, and SIV can all share.

Reply:
Yes, I plan to do it. But I am not sure about naming. Any suggestions?

Reply:
Personally I'd just call it `dbl`

Reply:
And trait and method names? Leave as is?

Reply:
The function is called "dbl" in all of the relevant papers. That said I'm not sure `Dblable` is a good name 😜

Reply:
I've published `dbl` crate and updated `pmac` to use it.

Reply:
Awesome. I'll use it in Miscreant for SIV as well